### PR TITLE
feat(ko): support multiple repositories

### LINKED ISSUE #5351: Add `repositories` key to `kos` configuration
### Is your feature request related to a problem? Please describe.

I've got a project I want to use `kos` on, and the project needs to publish to three different docker repositories. This shouldn't require building the image three times.

### Describe the solution you'd like

It'd be nice if there was a `repositories` key on `kos` rather than a `repository` and let it be a list.

### Describe alternatives you've considered

In the meantime I can work around just by duplicating the `kos` configuration.

### Search

- [X] I did search for other open and closed issues before opening this

### Supporter

- [X] I am a [sponsor](https://github.com/sponsors/caarlos0/) or a [Pro](https://goreleaser.com/pro) customer

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by caarlos0**
- allows to set more than one repository on KOs
- if many are set, build the first one, and use crane to copy to the next ones, preventing multiple builds

closes #5351

### CODE REVIEW

File: internal/pipe/ko/ko.go [hunk 1]
```diff
@@ -125,11 +127,16 @@ func (Pipe) Default(ctx *context.Context) error {
...
+		}
+
 		if repo := ctx.Env["KO_DOCKER_REPO"]; repo != "" {
-			ko.Repository = repo
+			ko.Repositories = []string{repo}
 		}
 
-		if ko.Repository == "" {
+		if len(ko.Repositories) == 0 {
 			return errNoRepository
```
Comment (alexandear) [inline internal/pipe/ko/ko.go:140]: I guess, we should change also the message text for `errNoRepository`. Currently, it mentions `repository` field:

```go
errNoRepository = errors.New("ko: missing repository: please set either the repository field or a $KO_DOCKER_REPO environment variable")
```

File: internal/pipe/ko/ko.go [hunk 2]
```diff
@@ -493,3 +511,33 @@ func validateMainPath(path string) error {
...
+func copyImage(src, dst string) (string, error) {
+	log.WithField("src", src).
+		WithField("dst", dst).
+		Info("copying manifest")
+	if err := crane.Copy(src, dst, crane.WithAuthFromKeychain(keychain)); err != nil {
+		return "", fmt.Errorf("ko: could not copy %q to %q: %w", src, dst, err)
+	}
+	digest, err := crane.Digest(dst, crane.WithAuthFromKeychain(keychain))
+	if err != nil {
+		return "", fmt.Errorf("ko: could get digest of %q: %w", dst, err)
```
Comment (alexandear) [inline internal/pipe/ko/ko.go:524]: I believe here is missed `not`:
```suggestion
		return "", fmt.Errorf("ko: could not get digest of %q: %w", dst, err)
```

File: www/docs/customization/ko.md [hunk 3]
```diff
@@ -52,9 +52,20 @@ kos:
     # The default user the image should be run as.
     user: "1234:1234"
 
+    # Repositories to push to.
+    #
+    # First one will be used on Ko build, the other ones will be copied from the
+    # first one using crane.
+    #
+    # Default: [ '$KO_DOCKER_REPO' ].
+    repository:
```
Comment (alexandear) [inline www/docs/customization/ko.md:61]: Looks like a copy-paste error:
```suggestion
    repositories:
```

File: internal/pipe/ko/ko.go (hunk 1 above)
Comment (caarlos0) [reply internal/pipe/ko/ko.go]: good call! thanks!

**Comment by tstirrat15** [comment]
:tada: thank you!

**Pull request merged**